File: 그리디/program_큰수만들기.java
import java.util.*;

class Solution {
    public String solution(String number, int k) {
        char[] nums = new char[number.length()];
        number.getChars(0, number.length(), nums, 0);
        Stack<Integer> stack = new Stack<>();
        int cnt = 0;
        for(char num : nums){
            Integer n = num - '0';
            while(stack.empty()==false && stack.peek()<n && cnt<k){
                stack.pop();
                cnt+=1;
            }
            stack.push(n);
        }
        while(cnt<k){
            stack.pop();
            cnt+=1;
        }
        String answer = "";
        for(Integer num : stack){
            answer += num.toString();
        }
        return answer;
    }
}

File: 그리디/program_큰수만들기.py
def solution(number, k):
    stack = []
    cnt = 0
    for x in number:
        temp = int(x)
        while stack and stack[-1]<temp and cnt<k:
            stack.pop()
            cnt+=1
        stack.append(temp)
    ## 나머지 처리
    if cnt<k:
        stack = stack[:(cnt-k)]
    answer = ''.join(map(str,stack))
    return answer